src/coreclr/jit/gentree.cpp

```diff
@@ -3835,10 +3835,17 @@ bool Compiler::gtIsLikelyRegVar(GenTree* tree)
 //
 bool Compiler::gtCanSwapOrder(GenTree* firstNode, GenTree* secondNode)
 {
-    // Relative of order of global / side effects can't be swapped.
-
     bool canSwap = true;
 
+    // Don't swap "CONST_HDL op CNS"
+    if (firstNode->IsIntegralConst() && secondNode->IsIntegralConst() && varTypeIsGC(firstNode) &&
+        !varTypeIsGC(secondNode))
```

Comment:
Do we need this last check?

Reply:
I can't think of a case of `CONST_HDL op CONST_HDL` so probably not

Reply:
Fixed